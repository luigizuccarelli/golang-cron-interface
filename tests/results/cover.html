
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">/home/lzuccarelli/Projects/myportfolio/myportfolio-cron/update.go (88.9%)</option>
				
				<option value="file1">/home/lzuccarelli/Projects/myportfolio/myportfolio-cron/validate.go (100.0%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">not covered</span>
				<span class="cov8">covered</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">package main

import (
        "bytes"
        "fmt"
        "io/ioutil"
        "net/http"
        "os"
        "strings"
        "time"
)

func UpdateData(c Client) error <span class="cov8" title="1">{

        list := strings.Split(os.Getenv("PAYLOAD"), ",")

        for x, _ := range list </span><span class="cov8" title="1">{
                // send the payload
                c.Info(fmt.Sprintf("Retrieving data for  %s", list[x]))
                req, _ := http.NewRequest("POST", os.Getenv("URL"), bytes.NewBuffer([]byte(list[x])))
                req.Header.Set("X-Api-Key", os.Getenv("APIKEY"))
                req.Header.Set("Content-Type", "application/json")
                resp, err := c.Do(req)
                if err != nil </span><span class="cov8" title="1">{
                        c.Error(fmt.Sprintf("Http request %v", err))
                        continue</span>
                }

                <span class="cov8" title="1">defer resp.Body.Close()
                body, e := ioutil.ReadAll(resp.Body)
                if e != nil </span><span class="cov0" title="0">{
                        c.Error(fmt.Sprintf("Cron service updatePrices %v", e))
                        continue</span>
                }
                <span class="cov8" title="1">c.Debug(fmt.Sprintf("Response from server %s", string(body)))
                time.Sleep(2 * time.Second)</span>
        }

        <span class="cov8" title="1">return nil</span>
}
</pre>
		
		<pre class="file" id="file1" style="display: none">package main

import (
        "errors"
        "fmt"
        "os"
        "strconv"
        "strings"

        "github.com/microlib/simple"
)

// checkEnvars - private function, iterates through each item and checks the required field
func checkEnvar(item string, logger *simple.Logger) error <span class="cov8" title="1">{
        name := strings.Split(item, ",")[0]
        required, _ := strconv.ParseBool(strings.Split(item, ",")[1])
        if os.Getenv(name) == "" </span><span class="cov8" title="1">{
                if required </span><span class="cov8" title="1">{
                        logger.Error(fmt.Sprintf("%s envar is mandatory please set it", name))
                        return errors.New(fmt.Sprintf("%s envar is mandatory please set it", name))
                }</span> else<span class="cov8" title="1"> {
                        logger.Error(fmt.Sprintf("%s envar is empty please set it", name))
                }</span>
        }
        <span class="cov8" title="1">return nil</span>
}

// ValidateEnvars : public call that groups all envar validations
// These envars are set via the openshift template
// Each microservice will obviously have a diffefrent envars so change where needed
func ValidateEnvars(logger *simple.Logger) error <span class="cov8" title="1">{
        items := []string{
                "LOG_LEVEL,false",
                "URL,true",
                "SLEEP,true",
                "CRON,true",
                "PAYLOAD,true",
        }
        for x, _ := range items </span><span class="cov8" title="1">{
                if err := checkEnvar(items[x], logger); err != nil </span><span class="cov8" title="1">{
                        return err
                }</span>
        }
        <span class="cov8" title="1">return nil</span>
}
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
