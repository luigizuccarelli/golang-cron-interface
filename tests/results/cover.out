mode: set
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/validator/validate.go:14.59,17.27 3 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/validator/validate.go:25.2,25.12 1 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/validator/validate.go:17.27,18.15 1 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/validator/validate.go:18.15,21.4 2 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/validator/validate.go:21.9,23.4 1 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/validator/validate.go:31.50,45.26 2 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/validator/validate.go:50.2,50.12 1 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/validator/validate.go:45.26,46.54 1 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/validator/validate.go:46.54,48.4 1 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/handlers/zip-transfer.go:17.48,23.66 4 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/handlers/zip-transfer.go:29.2,33.16 4 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/handlers/zip-transfer.go:38.2,42.16 5 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/handlers/zip-transfer.go:46.2,53.14 3 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/handlers/zip-transfer.go:58.2,59.14 2 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/handlers/zip-transfer.go:64.2,65.14 2 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/handlers/zip-transfer.go:70.2,70.12 1 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/handlers/zip-transfer.go:23.66,27.3 3 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/handlers/zip-transfer.go:33.16,36.3 2 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/handlers/zip-transfer.go:42.16,44.3 1 0
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/handlers/zip-transfer.go:53.14,56.3 2 1
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/handlers/zip-transfer.go:59.14,62.3 2 0
gitea-devops-shared-threefld-cicd.apps.c4.us-east-1.dev.aws.ocp.14west.io/cicd/trackmate-cron-interface/pkg/handlers/zip-transfer.go:65.14,68.3 2 0
